test/OpenTelemetry.Tests/Metrics/MeterProviderTests.cs

```diff
@@ -14,6 +14,8 @@
 // limitations under the License.
 // </copyright>
 
+using System.Reflection;
```

Comment:
Hmm, do we want to test against reflections?

Reply:
Fair enough I removed the test. But I would say anytime we learn of someone taking a dependency via reflection (which might be through such a PR) we should a) give a better mechanism so it isn't needed and b) [try to be nice as long as we can](https://github.com/open-telemetry/opentelemetry-dotnet/blob/a878ca3eead7d1b84c3e4c379c81b665b553c41b/src/OpenTelemetry/Logs/LogRecord.cs#L46) 🤣